src/core/ledger/policies/immediate.test.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import * as G from "../grain";
+import {processIdentities} from "../processedIdentities";
+import {random as randomUuid} from "../../../util/uuid";
+import {fromGrain} from "../nonnegativeGrain";
+import {immediateReceipts} from "./immediate";
+
+describe("core/ledger/policies/immediate", () => {
+  describe("immediateReceipts", () => {
+    const id1 = randomUuid();
+    const id2 = randomUuid();
+    const identities = [
+      {
+        cred: [10, 20, 30, 40],
+        id: id1,
+        paid: G.ONE,
+      },
+      {
+        cred: [200, 0, 0, 0],
+        id: id2,
+        paid: G.ZERO,
+      },
+    ];
+    const processedIdentities = processIdentities(identities);
+
+    it("errors on invalid range", () => {
+      const policy = {
+        policyType: "IMMEDIATE",
+        budget: fromGrain(G.ONE),
+        numPeriodsLookback: -1,
+      };
+      expect(() => immediateReceipts(policy, processedIdentities)).toThrowError(
+        `numPeriodsLookback must be at least 1`
+      );
+    });
+
+    it("errors on float instead of int", () => {
+      const policy = {
+        policyType: "IMMEDIATE",
+        budget: fromGrain(G.ONE),
+        numPeriodsLookback: 1.5,
+      };
+      expect(() => immediateReceipts(policy, processedIdentities)).toThrowError(
+        `numPeriodsLookback must be an integer`
+      );
+    });
+
+    it("defaults lookback period > history to max history", () => {
+      const policy1 = {
+        policyType: "IMMEDIATE",
+        budget: fromGrain(G.ONE),
+        numPeriodsLookback: 4,
+      };
+      const policy2 = {
+        policyType: "IMMEDIATE",
+        budget: fromGrain(G.ONE),
+        numPeriodsLookback: 50,
+      };
+      const expected1 = immediateReceipts(policy1, processedIdentities);
+      const expected2 = immediateReceipts(policy2, processedIdentities);
```

Comment:
nit technically this is `actual`

Reply:
Done, I wrote as follows:
```js
const expected = immediateReceipts(policy1, processedIdentities);
const actual = immediateReceipts(policy2, processedIdentities);
```
Of course both are actual, but since the test looks for the second to match the first, the first is essentially the "expected".